### ISSUE TITLE
AEM installation through maven content package plugin

### ISSUE THREAD

**Issue by joaotab** [unic/neba#149]
Hi,

   I've had some trouble interpreting the documentation regarding the deployment of the neba-delivery-aem package through maven:

http://neba.io/download.html

   It mentions the content package maven plugin but it is not very clear how to include it in the project structure generated by the AEM maven archetype.

   I've tried two separate approaches:

1) Create an entirely separate project for the NEBA delivery
2) Include an additional execution step in the "ui.apps" package

   In both cases I've been unable to work with the provided sample configuration, it seems it is lacking the version tag. After including the version it works, otherwise it doesn't seem to resolve the artifact. Also the dependency part mentioned in the doc doesn't seem to be needed, altough it is standard practice.

1) separate project

			<plugin>
				<groupId>com.day.jcr.vault</groupId>
				<artifactId>content-package-maven-plugin</artifactId>
				<executions>
					<execution>
						<configuration>
							<groupId>io.neba</groupId>
							<artifactId>io.neba.neba-delivery-aem</artifactId>
							<packaging>zip</packaging>
							**<version>${lib.neba.delivery.aem.version}</version>**
							<targetURL>http://${aem.host}:${aem.port}/crx/packmgr/service.jsp</targetURL>
							<failOnError>true</failOnError>
						</configuration>

						<goals>
							<goal>install</goal>
						</goals>
					</execution>
				</executions>
			</plugin>

2) "ui.apps"

                   <plugin>
                        <groupId>com.day.jcr.vault</groupId>
                        <artifactId>content-package-maven-plugin</artifactId>
                        <executions>
                            <execution>
                                <id>install-package</id>
                                <goals>
                                    <goal>install</goal>
                                </goals>
                            </execution>
							<execution>
								<id>install-neba</id>
								<configuration>
									<groupId>io.neba</groupId>
									<artifactId>io.neba.neba-delivery-aem</artifactId>
									<packaging>zip</packaging>
									**<version>${lib.neba.delivery.aem.version}</version>**
								</configuration>		
								<goals>
									<goal>install</goal>
								</goals>
							</execution>                            
                        </executions>
                        <configuration>
                            <targetURL>http://${aem.host}:${aem.port}/crx/packmgr/service.jsp</targetURL>
                            <failOnError>true</failOnError>
                            <failOnMissingEmbed>true</failOnMissingEmbed>
                        </configuration>
                     </plugin>

Also may be worth mentioning that AEM 6.1 SP2 solves the issue #69 with javax.inject being incorrectly exported by Sling Models / Java Scripting API bundles, we had to include the geronimo spec bundle for providing the javax.inject dependency through the "embedded" section in the content package plugin.

Cheers, João

**Comment by olaf-otto** [comment]
Hi @joaotab , thanks a lot for the hints and testing effort! 

Indeed, We've not yet thought about a best-practice of integrating into the Adobe's default AEM archetype.

Regardless, I have updated the download docs with regard to your findings and - hopefully - made the part with the dependencies more clear. Could you please review?

Regarding the AEM SP2: I believe the issue is still existing in AEM 6.2 at the moment, but indeed we may be able to remove the customized version range in NEBA. At the moment, NEBA will accept any javax.inject package in the range `[0,2)`. 

I didn't get what you meant regarding the deployment of the javax.inject geronimo spec bundle via the embedded dependencies - was this something your own app was depending on?

**Comment by joaotab** [comment]
Hi @olaf-otto, thanks, I see you've included the version in the doc. My original issue was that I had to dig around to find out why it wasn't being deployed and it was about the missing version.

Regarding the inject, it is in fact related with the sample code from the AEM archetype, I just realized we still have some of it lying around. We've switched to resource and autowired for osgi and spring wiring in the meantime.
